extensions/shared/external-media/sources/with-media.js

```diff
@@ -53,6 +54,42 @@ export default function withMedia() {
 				};
 			}
 
+			modalRef = el => {
+				if ( el ) {
+					// Find the modal wrapper.
+					this.modalElement = el.closest( '.jetpack-external-media-browser' );
+
+					// Attach the listener if found.
+					if ( this.modalElement ) {
+						this.modalElement.addEventListener( 'keydown', this.stopArrowKeysPropagation );
+					}
+				} else if ( this.modalElement ) {
+					// Remove listeners when unmounting.
+					this.modalElement.removeEventListener( 'keydown', this.stopArrowKeysPropagation );
+					this.modalElement = null;
```

Comment:
Can you explain how this works? Does `modalRef` get called again on unmount? And at that point, the `el` would be undefined since it's unmounted, but `this.modalElement` would still exist, thus the event listener would get removed?

Reply:
Confirmed this behavior when testing. Works well!

Reply:
You got it. Ref gets called after the DOM element gets mounted and the element is passed as an argument. The next time it is called, it is just before unmounting with an empty argument that tells you to clean up.